routing/router.go

```diff
@@ -345,6 +351,44 @@ func (r *ChannelRouter) syncGraphWithChain() error {
 		return nil
 	}
 
+	// If the main chain blockhash at prune height is different from the
+	// prune hash, this might indicate the database is on a stale branch.
+	mainBlockHash, err := r.cfg.Chain.GetBlockHash(int64(pruneHeight))
+	if err != nil {
+		return err
+	}
+
+	// While we are on a stale branch of the chain, walk backwards to find
+	// first common block.
+	for !pruneHash.IsEqual(mainBlockHash) {
+		log.Infof("channel graph is stale. Disconnecting block %v "+
+			"(hash=%v)", pruneHeight, pruneHash)
+		// Prune the graph for every channel that was opened at height
+		// >= pruneHeigth.
+		if err := r.cfg.Graph.DisconnectBlockAtHeight(pruneHeight); err != nil {
+			return err
+		}
+
+		pruneHash, pruneHeight, err = r.cfg.Graph.PruneTip()
+		if err != nil {
+			switch {
+			// If at this point the graph has never been pruned, we
+			// break out of the loop, as the database is a blank
+			// slate.
+			case err == channeldb.ErrGraphNeverPruned:
+				break
+			case err == channeldb.ErrGraphNotFound:
+				break
+			default:
+				return err
+			}
+		}
+		mainBlockHash, err = r.cfg.Chain.GetBlockHash(int64(pruneHeight))
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
The logging statement below should be modified to use `mainBlockHash` and height instead, as in the case we re-org'd back, the statement wouldn't be accurate.

Reply:
I don't think it's necessary, since if we reorg'd back, then `pruneHash == mainBlockHash` and same for height, which should make it accurate.

If I'm wrong, the probably the following loop also would need to change, as it loops from `pruneHeight` to `bestHeight`.

Reply:
Nope you're correct, I missed that both `pruneHash` and `pruneHeight` were re-assigned within the loop above.